### ISSUE TITLE
Feature request: targets.docs.deps and targets.benchmark.deps

### ISSUE THREAD

**Issue by tkf** [JuliaLang/Pkg.jl#521]
It would be nice to have `targets.docs.deps` and `targets.benchmark.deps`, etc. (or `docs/Project.toml` and `benchmark/Project.toml` somehow stacked on top of the root `Project.toml`) for document- and benchmark- specific dependencies.  Are you planning to add such features?

It may be related to #480 since I guess those features will share the implementation.

**Comment by StefanKarpinski** [comment]
You can add these deps to `[extras]` and then have `targets.banchmarks = [...]` to list the packages you want in this target. What do you imagine that this target would mean to the package manager?

**Comment by tkf** [comment]
Yes, I saw you wrote #531 so I thought it was solved and closed it (probably should have commented that).

> What do you imagine that this target would mean to the package manager?

I thought the best answer would be *nothing* in the sense that the environments for `./docs`, `./benchmark` and `./test` are yet another environment with extra packages on top of `[deps]` and the package manager doesn't care what it's for.  From what I understand from the PR, it looks like your design of `[extras]` and `[targets]` covers arbitrary targets so that looks great.  What is not clear for me at the moment is the way to activate one of `[targets]` (as already mentioned in https://github.com/JuliaLang/Pkg.jl/issues/480#issuecomment-406885738) but I suppose that's more like `julia` issue?  Ideally, I'd like to do something like `pkg> activate @:docs` in REPL or `julia --project=@:docs docs/make.jl` from shell to activate `docs` target in current project `@` (completely made-up syntax).
